core/pouch/index.js

```diff
@@ -471,6 +471,30 @@ class Pouch {
     return this.updater.push(task, callback)
   }
 
+  async unsyncedDocIds() {
+    const localSeq = await this.getLocalSeqAsync()
+    return new Promise((resolve, reject) => {
+      this.db
+        .changes({
+          since: localSeq,
+          filter: '_view',
+          view: 'byPath'
+        })
+        .on('complete', ({ results }) => resolve(results.map(r => r.id)))
+        .on('error', err => reject(err))
+    })
+  }
+
+  // Touch existing documents with the given ids to make sure they appear in the
+  // changesfeed.
+  // Careful: this will change their _rev value!
+  async touchDocs(ids /*: string[] */) {
+    const existingDocs = await Promise.all(
+      ids.map(id => this.byIdMaybeAsync(id))
+    ).filter(d => d)
```

Comment:
I think it would be better to fetch the documents with a call to `allDocs` with a keys option: it makes a single call to pouchdb instead of a call per document.